### PR TITLE
[RDY] Add an ExitCommand in Eclipse

### PR DESCRIPTION
**Pull request by Hjdskes**

### CODE REVIEW

**Comment by Unia** [comment]
The "Open in REPL" doesn't work yet as I have to get it inside the language's submenu in the project explorer's context menu and I cannot in advance know those IDs for all the languages. Example below:

image

File: org.metaborg.spoofax.shell.eclipse/plugin.xml [hunk 1]
```diff
@@ -13,4 +13,30 @@
...
+        <command id="org.metaborg.spoofax.shell.client.eclipse.handlers.open" name="Open in REPL"/>
+    </extension>
+    <extension point="org.eclipse.ui.handlers">
+        <handler commandId="org.metaborg.spoofax.shell.client.eclipse.handlers.open"
+                     class="org.metaborg.spoofax.shell.client.eclipse.handlers.OpenHandler"/>
+    </extension>
+
+    <extension point="org.eclipse.ui.menus">
+        <!-- Menu item in the project context menu (package- and project explorer). -->
+        <menuContribution locationURI="popup:org.metaborg.spoofax.eclipse.menu.project">
```
Comment (Hjdskes) [inline org.metaborg.spoofax.shell.eclipse/plugin.xml]: This section is where I need to register the button in all the submenus of all the languages.

File: org.metaborg.spoofax.shell.eclipse/plugin.xml [hunk 2]
```diff
@@ -13,4 +13,30 @@
...
+            <command commandId="org.metaborg.spoofax.shell.client.eclipse.open"
```
Comment (Hjdskes) [inline org.metaborg.spoofax.shell.eclipse/plugin.xml]: Typo, must be `"org.metaborg.spoofax.shell.client.eclipse.handlers.open"`

File: org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseRepl.java [hunk 3]
```diff
@@ -24,12 +27,16 @@
 /**
  * An Eclipse-based implementation of {@link IRepl}.
  *
- * It uses a multiline input editor with keyboard shortcuts, including persistent history, syntax
- * highlighting and error marking.
+ * It uses a multiline input editor with keyboard shortcuts, including volatile history error
```
Comment (Balletie) [inline org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseRepl.java]: > and error marking

File: org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseRepl.java [hunk 4]
```diff
@@ -88,10 +95,17 @@ private void runAsJob(final String input) {
             @Override
             protected IStatus run(IProgressMonitor monitor) {
                 try {
-                    IResult result = pool.submit(() -> eval(input)).get();
+                    IResult result =
+                        pool.submit(() -> eval(input)).get(TIMEOUT, TIMEUNIT);
```
Comment (Balletie) [inline org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseRepl.java]: What does this fix?

Comment (Hjdskes) [inline org.metaborg.spoofax.shell.eclipse/src/main/java/org/metaborg/spoofax/shell/client/eclipse/impl/EclipseRepl.java]: Infinite loops

**Comment by Hjdskes** [comment]
Don't merge this yet; it throws an NPE in `EclipseDisplay#displayStyledText` because of line 108 in `EclipseRepl`.

**Comment by Hjdskes** [comment]
^ Fixed.

**Pull request merged**